pulley/src/interp/tail_loop.rs

```diff
@@ -59,20 +64,47 @@ macro_rules! tail_call {
 }
 
 impl Interpreter<'_> {
-    pub fn run(mut self) -> Done {
-        // Perform a dynamic dispatch through a function pointer indexed by
-        // opcode.
-        let opcode = unwrap_uninhabited(Opcode::decode(self.bytecode()));
-        let handler = OPCODE_HANDLER_TABLE[opcode as usize];
-        tail_call!(handler(self));
+    pub fn run(self) -> Done {
+        dispatch(self.state, self.pc, self.executing_pc)
```

Comment:
I'll note that this change is required to appease `become` because `run`'s signature differed from the handler's signature (`Interpreter`-by-value vs exploded-to-components)